pkg/volume/csi_migration.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"errors"
+	"fmt"
+
+	"k8s.io/api/core/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-base/featuregate"
+	csilibplugins "k8s.io/csi-translation-lib/plugins"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+// InTreeToCSITranslator contains methods required to check migratable status
+// and perform translations from InTree PV's to CSI
+type InTreeToCSITranslator interface {
+	GetInTreePluginNameFromSpec(pv *v1.PersistentVolume, vol *v1.Volume) (string, error)
+	TranslateInTreePVToCSI(pv *v1.PersistentVolume) (*v1.PersistentVolume, error)
+	TranslateInTreeInlineVolumeToCSI(vol *v1.Volume) (*v1.PersistentVolume, error)
+}
+
+// CSIMigratedPluginManager keeps track of migrated state of in-tree plugins
+type CSIMigratedPluginManager struct{
+	translator InTreeToCSITranslator
+}
+
+// NewCSIMigratedPluginManager returns a new CSIMigratedPluginManager instance
+func NewCSIMigratedPluginManager(t InTreeToCSITranslator) CSIMigratedPluginManager {
+	return CSIMigratedPluginManager{
+		translator: t,
+	}
+}
+
+func isFeatureFlagEnabled(key featuregate.Feature) bool {
```

Comment:
I think we should remove this helper function - it's not really making anything much simpler and kind of obscures the semantics used elsewhere in k8s/k8s.